crates/pallet-domains/src/block_tree.rs

```diff
@@ -881,36 +881,38 @@ mod tests {
             let head_node = get_block_tree_node_at::<Test>(domain_id, head_receipt_number).unwrap();
             assert_eq!(head_node.operator_ids, operator_set);
 
-            // Prepare the inbainvalid bundles and bundle authors
-            let invalid_bundle_authors: Vec<_> = operator_set
-                .clone()
-                .into_iter()
-                .filter(|i| i % 2 == 0)
-                .collect();
-            let invalid_bundles = invalid_bundle_authors
-                .iter()
-                .map(|i| InvalidBundle {
-                    // operator id start at 1 while bundle_index start at 0, thus minus 1 here
-                    bundle_index: *i as u32 - 1,
-                    invalid_bundle_type: InvalidBundleType::OutOfRangeTx,
-                })
-                .collect();
-
-            // Get the `bunde_extrinsics_roots` that contains all the submitted bundles
+            // Get the `bundles_extrinsics_roots` that contains all the submitted bundles
             let current_block_number = frame_system::Pallet::<Test>::current_block_number();
             let execution_inbox = ExecutionInbox::<Test>::get((
                 domain_id,
                 current_block_number,
                 current_block_number,
             ));
-            let bunde_extrinsics_roots: Vec<_> = execution_inbox
+            let bundles_extrinsics_roots: Vec<_> = execution_inbox
                 .into_iter()
                 .map(|b| b.extrinsics_root)
                 .collect();
-            assert_eq!(bunde_extrinsics_roots.len(), operator_set.len());
+            assert_eq!(bundles_extrinsics_roots.len(), operator_set.len());
 
-            target_receipt.invalid_bundles = invalid_bundles;
-            target_receipt.block_extrinsics_roots = bunde_extrinsics_roots;
+            // Prepare the invalid bundles and invalid bundle authors
+            let mut bundles = vec![];
+            let mut invalid_bundle_authors = vec![];
+            for (i, (operator, extrinsics_root)) in operator_set
+                .iter()
+                .zip(bundles_extrinsics_roots)
+                .enumerate()
+            {
+                if i % 2 == 0 {
```

Comment:
What does this do?

Reply:
2 is a magic number to generate both valid and invalid bundles for testing.